### PR TITLE
az load support for zip artifacts upload

### PR DESCRIPTION
**Pull request by mbhardwaj-msft**
---
### Changes
- vendored_sdk upgrade for ZIP artifacts
- az load CLI support for ZIP artifacts upload to a test
- fixes on return types post typespec generated sdk upgrade
- bug fix to allow relative path of files to upload when load test create with config file
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml
Comment (mitsha-microsoft) [inline src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml:6]: Can you put a comment on why this config is invalid?

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py:615]: I see, I was wondering why there wasn't validation check, will remove

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py:665]: Got it, but the validation check applies for the ZIP files, right?

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py:616]: The deletion logic is slightly different in case of user_properties and test_plan and test_plan is a different object in itself, so didn't use the upload_generic_files_helper for these two because handling these two differently increases the complexity of the method and we'll be back to square one
image

File: src/load/azext_load/tests/latest/test_load_testrun.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/tests/latest/test_load_testrun.py]: removed, got left after some local testing

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py]: updated order of file upload

File: src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml
Comment (mbhardwaj-msft) [reply src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml:7]: added comments stating reason

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py]: removed for configuration files as well, continues to stay for the zipped artifacts

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mitsha-microsoft) [reply src/load/azext_load/data_plane/utils/utils.py]: Got it, was trying to reduce code duplication. See if there's some possibility around here

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mitsha-microsoft) [reply src/load/azext_load/data_plane/utils/utils.py]: Yeah. We validate test plans and zip files, we dont validate other artifacts, hence there were no checks. (I think we should sort of unify or make the check generic to handle terminal validation status)

File: src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml
Comment (mitsha-microsoft) [reply src/load/azext_load/tests/latest/resources/invalid-config-with-zip-artifacts.yaml:7]: Thanks!

**Review by mitsha-microsoft — APPROVED** [review]
LGTM

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by mbhardwaj-msft** [comment]
> Please fix CI issues

I have an ongoing PR from Sept which fixes the CI issues, waiting on its review and approval https://github.com/Azure/azure-cli-extensions/pull/8014

File: src/load/azext_load/data_plane/utils/utils.py
Comment (mbhardwaj-msft) [reply src/load/azext_load/data_plane/utils/utils.py]: refactored a bit by defining what `existing_files` constitute for the various file types

File: src/load/HISTORY.rst
Comment (mitsha-microsoft) [inline src/load/HISTORY.rst:11]: Think we can remove the mention for internal refactor etc. Is it done for other extensions?

**Review by mitsha-microsoft — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**
